code/go/0chain.net/smartcontract/rest/rest.go

```diff
@@ -55,6 +56,22 @@ func NewRestHandler(c QueryChainer) RestHandlerI {
 
 func (rh *RestHandler) Register(endpoints []Endpoint) {
 	for _, e := range endpoints {
-		http.HandleFunc(e.URI, e.Handler)
+		http.HandleFunc(e.URI, WithCORS(e.Handler))
+	}
+}
+
+// WithCORS enable CORS
+func WithCORS(fn func(w http.ResponseWriter, r *http.Request)) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*") // CORS for all.
```

Comment:
What's the purpose for adding CORS while we allowing all possible origin.

Reply:
NVM, it should be used by browser to run Apps that send request to the blockchain network.